lasio/las.py

```diff
@@ -704,6 +704,43 @@ def set_data_from_df(self, df, **kwargs):
                 str(name) for name in df.columns.values
             ]
         self.set_data(df_values, **kwargs)
+    
+    def stack_curves(self, curve_list=None, stub=None, sort_curves=True):
+        """Stack multi-channel curve data to a numpy 2D ndarray. Provide a 
+        list of curves or a stub name (prefix shared by all curves that will
+        be stacked). If `stub` and `curve_list` are both supplied, `stub`
+        is used.
+        
+        Keyword Arguments:
+            stub (str): Supply the first several characters of the channel
+                set to be stacked. 
+            curve_list (list): Or a list of the curves to be stacked
+            sort_curves (bool): Natural sort curves based on mnemonic prior 
+                to stacking.
+
+        Returns:
+            2-D numpy array
+        """
+        curves = self.curves.keys()
+        if stub:
+            channels = [i for i in curves if i.startswith(stub)]
+        elif curve_list:
+            channels = curve_list
+        if sort_curves:
+            channels.sort(key=self._sort_channels)
+        if channels:
```

Comment:
Instead of if blocks here and at line 733 I feel we should assume that channels and therefore indices are found. If not the function should raise an appropriate exception, rather than returning None. We don't necessarily need to raise it but hopefully it will raise something sensible if we drop the if blocks.

Also that case - where channels are not found - should also be a test.

Reply:
Fair point! I decided to raise some built-in exceptions manually because the method wasn't raising "something sensible" in a couple cases where I thought it should. I hope the new implementation isn't too much.